--- nvml.h	2020-09-24 12:43:36.060860551 +0000
+++ nvml.patched.h	2020-09-24 12:45:41.412702233 +0000
@@ -128,7 +128,9 @@
  */
 #define NVML_VALUE_NOT_AVAILABLE (-1)
 
-typedef struct nvmlDevice_st* nvmlDevice_t;
+typedef struct {
+	struct nvmlDevice_st* handle;
+} nvmlDevice_t;
 
 /**
  * Buffer size guaranteed to be large enough for pci bus id
@@ -1235,7 +1237,9 @@
  */
 /***************************************************************************************************/
 
-typedef struct nvmlUnit_st* nvmlUnit_t;
+typedef struct {
+	struct nvmlUnit_st* handle;
+} nvmlUnit_t;
 
 /**
  * Description of HWBC entry
@@ -1337,7 +1341,9 @@
 /**
  * Handle to an event set
  */
-typedef struct nvmlEventSet_st* nvmlEventSet_t;
+typedef struct {
+	struct nvmlEventSet_st* handle;
+} nvmlEventSet_t;
 
 /** @defgroup nvmlEventType Event Types
  * @{
@@ -6956,7 +6962,9 @@
     nvmlGpuInstancePlacement_t placement; //!< Placement for this instance
 } nvmlGpuInstanceInfo_t;
 
-typedef struct nvmlGpuInstance_st* nvmlGpuInstance_t;
+typedef struct {
+	struct nvmlGpuInstance_st* handle;
+} nvmlGpuInstance_t;
 
 /**
  * Compute instance profiles.
@@ -6995,7 +7003,9 @@
     unsigned int profileId;        //!< Unique profile ID within the GPU instance
 } nvmlComputeInstanceInfo_t;
 
-typedef struct nvmlComputeInstance_st* nvmlComputeInstance_t;
+typedef struct {
+	struct nvmlComputeInstance_st* handle;
+} nvmlComputeInstance_t;
 
 /**
  * Set MIG mode for the device.
